lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,353 @@
+          type: int
+          sample: 514
+        protocol:
+          description: The network protocol of the server.
+          returned: always
+          type: str
+          sample: "udp"
+        stime:
+          description: The start time of the listening process.
+          returned: always
+          type: str
+          sample: "Thu Feb  2 13:29:45 2017"
+        user:
+          description: The user who is running the listening process.
+          returned: always
+          type: str
+          sample: "root"
+'''
+
+import re
+import platform
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+
+
+def netStatParse(raw):
+    results = list()
+    for line in raw.splitlines():
+        listening_search = re.search('[^ ]+:[0-9]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('([^ ]+):([0-9]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                protocol = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                protocol = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('(([0-9]+)/(.*)|-)', pidstr)
+            if conns and pids:
+                address = conns.group(1)
+                port = conns.group(2)
+                if (pids.group(2)):
+                    pid = pids.group(2)
+                else:
+                    pid = 0
+                if (pids.group(3)):
+                    name = pids.group(3)
+                else:
+                    name = ''
+                result = dict(pid=int(pid), address=address, port=int(port), protocol=protocol, name=name)
+                if result not in results:
+                    results.append(result)
+            else:
+                raise EnvironmentError('Could not get process information for the listening ports.')
+    return results
+
+
+def applyWhitelist(portspids, whitelist=None):
+    if whitelist is None:
+        whitelist = list()
+    processes = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            process = dict(pid=p['pid'], address=p['address'], port=p['port'], protocol=p['protocol'], name=p['name'], stime=p['stime'], user=p['user'])
+            if process not in processes:
+                processes.append(process)
+    return processes
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            whitelist_tcp=dict(type='list', default=list()),
+            whitelist_udp=dict(type='list', default=list()),
+        ),
+        supports_check_mode=True,
+    )
+
+    if platform.system() != 'Linux':
+        module.fail_json(msg='This module requires Linux.')
+
+    def getPidSTime(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'lstart', '-p', str(pid)])
```

Comment:
I don't think this needs to be converted into a string.
```suggestion
        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'lstart', '-p', pid])
```

Reply:
Seems not casting to a string raises a TypeError:

https://app.shippable.com/github/ansible/ansible/runs/113828/47/console

```
Traceback (most recent call last):
  File \"/root/.ansible/tmp/ansible-tmp-1552552468.29-100247799520006/AnsiballZ_listen_ports_facts.py\", line 134, in <module>
    _ansiballz_main()
  File \"/root/.ansible/tmp/ansible-tmp-1552552468.29-100247799520006/AnsiballZ_listen_ports_facts.py\", line 126, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/root/.ansible/tmp/ansible-tmp-1552552468.29-100247799520006/AnsiballZ_listen_ports_facts.py\", line 69, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File \"/tmp/ansible_listen_ports_facts_payload_3292BE/__main__.py\", line 353, in <module>
  File \"/tmp/ansible_listen_ports_facts_payload_3292BE/__main__.py\", line 328, in main
  File \"/tmp/ansible_listen_ports_facts_payload_3292BE/__main__.py\", line 292, in getPidSTime
  File \"/tmp/ansible_listen_ports_facts_payload_3292BE/ansible_listen_ports_facts_payload.zip/ansible/module_utils/basic.py\", line 2725, in run_command
  File \"/usr/lib64/python2.6/posixpath.py\", line 283, in expandvars
    if '$' not in path:
TypeError: argument of type 'int' is not iterable

```

Reply:
Nevermind, carry on :-)